docs/src/main/asciidoc/security-basic-authentication.adoc

```diff
@@ -69,6 +69,33 @@ For more information about how you can secure your Quarkus applications by using
 {project-name} also includes built-in security to allow for role-based access control (RBAC) based on the common security annotations `@RolesAllowed`, `@DenyAll`, `@PermitAll` on REST endpoints and CDI beans.
 For more information, see the Quarkus xref:security-authorize-web-endpoints-reference.adoc[Authorization of web endpoints] guide.
 
+[[basic-auth-programmatic-set-up]]
+== Set up Basic authentication programmatically
+
+The `io.quarkus.vertx.http.security.event.Basic` CDI event allows to customize the configuration provided in the 'application.properties' during the runtime like in the example below:
+
+[source,java]
+----
+package org.acme.http.security;
+
+import io.quarkus.vertx.http.security.event.Basic;
+import jakarta.enterprise.event.Observes;
+
+public class BasicAuthenticationConfiguration {
+
+    void configure(@Observes Basic basic) {
+        basic
+            .enable() <1>
+            .realm("QuarkusTestRealm"); <2>
+    }
+
+}
+----
+<1> Enable basic authentication.
+Downside is that Quarkus doesn't know that the basic authentication is required during the build-time.
```

Comment:
This probably should be moved to the `NOTE`, and instead of using a `downside` word,  clarify that using the build time configuration is needed for some cases like generating an OpenAPI schema, etc